src/python/gudhi/representations/vector_methods.py

```diff
@@ -285,70 +287,167 @@ def __call__(self, diag):
         """
         return self.fit_transform([diag])[0,:]
 
+
 class BettiCurve(BaseEstimator, TransformerMixin):
     """
-    This is a class for computing Betti curves from a list of persistence diagrams. A Betti curve is a 1D piecewise-constant function obtained from the rank function. It is sampled evenly on a given range and the vector of samples is returned. See https://www.researchgate.net/publication/316604237_Time_Series_Classification_via_Topological_Data_Analysis for more details.
+    Compute Betti curves from persistence diagrams. There are two modes of operation: with a predefined grid, and without. With a predefined grid, the class computes the Betti numbers at those grid points. Without a predefined grid, it can be fit to a list of persistence diagrams and produce a grid that consists of (at least) the filtration values at which at least one of those persistence diagrams changes Betti numbers, and then compute the Betti numbers at those grid points. In the latter mode, the exact Betti curve is computed for the entire real line.
+
+    Parameters
+    ----------
+    predefined_grid: 1d array, triple or None, default=None
+        Predefined filtration grid points at which to compute the Betti curves. Must be strictly ordered. Infinities are OK. If a triple of the form (l, u, n), the grid will be uniform from l to u in n steps. If None (default), a grid will be computed that captures all changes in Betti numbers in the provided data.
+
+    Attributes
+    ----------
+    grid_: 1d array
+        The grid on which the Betti numbers are computed. If predefined_grid was specified, grid_ will always be that grid, independently of data. If not, the grid is fitted to capture all filtration values at which the Betti numbers change.
+
+    Examples
+    --------
+    If pd is a persistence diagram and xs is a nonempty grid of finite values such that xs[0] >= pd.min(), then the result of
+    >>> bc = BettiCurve(xs)
+    >>> result = bc(pd)
+    and
+    >>> from scipy.interpolate import interp1d
+    >>> bc = BettiCurve(None)
+    >>> bettis = bc.fit_transform([pd])
+    >>> interp = interp1d(bc.grid_, bettis[0, :], kind="previous", fill_value="extrapolate")
+    >>> result = np.array(interp(xs), dtype=int)
+    are the same.
     """
-    def __init__(self, resolution=100, sample_range=[np.nan, np.nan]):
```

Comment:
I don't know if we want that, but it would be possible to avoid breaking users' code if we enabled the new exact mode with resolution=None or resolution="exact", instead of introducing a new parameter predefined_grid.

Reply:
Would it be ok with you if we changed the interface in such a way? Not breaking existing user code would make it easier for us to merge this branch.
We can do the change, or you can do it, as you prefer.

Reply:
Hi. Sorry for forgetting about this discussion. Feel free to change the interface to whatever you prefer – I have no strong feelings about it (just that there should be something à la the exact construction that I proposed available :-)).